packages/dds/tree/docs/main/compatibility.md

```diff
@@ -147,3 +147,83 @@ Snapshot tests are effective at catching changes which inadvertently modify the
 
 Tree2's full-scale snapshot tests can be found at [experimental/dds/tree2/src/test/snapshots/summary.spec.ts](../../src/test/snapshots/summary.spec.ts),
 with smaller-scale snapshot tests (e.g. snapshot testing just the SchemaIndex format) nearby.
+
+# Implementation Specifics
+
+SharedTree's codecs are frequently composed over each other in a manner consistent with SharedTree's layering.
+This conveniently allows unit testing only portions of the persisted data.
+As a downside, it makes maintaining backwards compatibility with the format somewhat more complex, because not all codecs in this composition provide explicit versions for the portion of data they encode.
+The following diagram shows _runtime dependencies_ of the codec hierarchy for SharedTree's original persisted format.
+
+```mermaid
+flowchart TD
+    RevisionTagCodec
+    SchemaCodec
+    FieldBatchCodec
+    ForestCodec-->FieldBatchCodec
+    MessageCodec-->ChangeFamilyCodec
```

Comment:
It's not obvious that `ChangeFamilyCodec` refers to the codec for `SharedTreeChange`. I wonder if it would be better to name the nodes after what they encode (which is a specific type in the code). I think it would make it easier for the reader to understanding the mapping between the diagram and the codebase. That said, such a naming scheme would break down if we were to have different independent codecs/codec families for a single format. I'm guessing that's unlikely though.

Reply:
I'm open to that. I had less descriptive names for these originally and came to a somewhat similar conclusion, though I didn't settle on an identifier--intent was you could search each term and pretty clearly find what it referred to. My worry with in-memory types is that they might be used in many other places. But it's probably true that a search for the in-memory type in files ending in `Codec/Codecs` turns up easily. I'll play around with it.

Reply:
I included a table mapping the name in the diagram to the file and in-memory type that the codec encodes. I think even if this goes out-of-date, that should help the reader determine the intent / how to update things.